napi-inl.h

```diff
@@ -1379,6 +1379,41 @@ inline void Object::AddFinalizer(Finalizer finalizeCallback,
   }
 }
 
+inline Object::iterator::iterator(const Object* object, const Type type) {
+  _object = object;
+  _keys = object->GetPropertyNames();
+  _index = type == Type::BEGIN ? 0 : _keys.Length();
+}
+
+inline Object::iterator Napi::Object::begin() {
+  iterator it(this, Object::iterator::Type::BEGIN);
+  return it;
+}
+
+inline Object::iterator Napi::Object::end() {
+  iterator it(this, Object::iterator::Type::END);
+  return it;
+}
+
+inline Object::iterator& Object::iterator::operator++() {
+  ++_index;
+  return *this;
+}
+
+inline bool Object::iterator::operator==(const iterator& other) const {
+  return _index == other._index;
+}
+
+inline bool Object::iterator::operator!=(const iterator& other) const {
+  return _index != other._index;
+}
+
+inline std::pair<Value, Value> Object::iterator::operator*() const {
+  const Value key = _keys.Get(_index);
+  const Value value = _object->Get(key);
+  return {key, value};
+}
```

Comment:
This is technically still a `const_iterator`. I don't know how I could return an `std::pair` of references to the key and the value. Some help please?